connector/spanmetricsconnector/connector.go

```diff
@@ -334,13 +340,31 @@ func (p *connectorImp) getOrCreateResourceMetrics(attr pcommon.Map) *resourceMet
 	return v
 }
 
+// contains checks if string slice contains a string value
+func contains(elements []string, value string) bool {
+	for _, element := range elements {
+		if value == element {
+			return true
+		}
+	}
+	return false
+}
+
 func (p *connectorImp) buildAttributes(serviceName string, span ptrace.Span, resourceAttrs pcommon.Map) pcommon.Map {
 	attr := pcommon.NewMap()
 	attr.EnsureCapacity(4 + len(p.dimensions))
-	attr.PutStr(serviceNameKey, serviceName)
-	attr.PutStr(spanNameKey, span.Name())
-	attr.PutStr(spanKindKey, traceutil.SpanKindStr(span.Kind()))
-	attr.PutStr(statusCodeKey, traceutil.StatusCodeStr(span.Status().Code()))
+	if !contains(p.config.ExcludeDimensions, serviceNameKey) {
```

Comment:
Could you add a test to verify that excluded attributes aren't added?

Reply:
Added in new commit as a separate test, cause from verifier function we do not have an access to the config to check what exactly was disabled.